contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -208,15 +208,35 @@ pltsql_createFunction(ParseState *pstate, PlannedStmt *pstmt, const char *queryS
 			Oid func_oid = InvalidOid;
 			char *schemaname = NULL;
 			char *funcname = NULL;
+			List *path_oids = NULL;
+			char *cur_schema_name = NULL;
 
 			func = makeNode(ObjectWithArgs);
-			func->objname = stmt->funcname;
-			func->args_unspecified = true;
 
 			/* Get the schema name and function name. */
 			DeconstructQualifiedName(stmt->funcname, &schemaname, &funcname);
 
-			/* function, procedure */
+			/* If schema name is not specified, use the current default schema */
+			if (schemaname == NULL || !strlen(schemaname))
+			{
+				path_oids = fetch_search_path(false);
+				if (path_oids != NIL)
```

Comment:
What if len(path_oids) > 1 ? In that case how do we decide? Is there any such case?

Reply:
basically `path_oids` returns the list of oids of the all schemas present in the active search path and we only want the current schema oid for our case.